packages/pnpm/src/err.ts

```diff
@@ -11,7 +11,11 @@ export function errorHandler (error: Error) {
     return
   }
   if (error.name != null && error.name !== 'pnpm' && !error.name.startsWith('pnpm:')) {
-    error.name = 'pnpm'
+    try {
+      error.name = 'pnpm'
+    } catch {
+      // Sometimes the name property is read-only
+    }
```

Comment:
I think this change can also be merged to v6 branch since there has a similar issue: https://github.com/pnpm/pnpm/issues/4270

Reply:
oh, wow